_posts/2018-11-13-type-constraints-correctness-reuse.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: Type constraints for code reuse and readability
+author: jguitana
+---
+
+The case for statically typed languages is so that we can prove our programs are right at compile time. Although our languages are not powerful enough to achieve this all the time, we certainly not always use their full capabilities. Our intuition should guide us to reduce implementation space and improve code reuse, allowing us to focus on small parts of a big problem. We can only do this if we choose the right abstraction in the form of Types, because these act as constraints on the implementation space, and if built correctly bring with them reusable and tested code.
```

Comment:
I'd recommend to end the sentence at "int the form of Types.". Then "These act..." and "and"s I think don't have commas right before.

Reply:
Having "and"'s after commas isn't incorrect, but I'd remove it in this case because IMO `if built correctly` should be between commas (which would mean a lot of commas).
Actually, I would rephrase the last part of the sentence for clarity:
` because these act as constraints on the implementation space and bring with them reusable and tested code, if built correctly.`

Reply:
Settled for a mix if you agree: `We can only do this if we choose the right abstraction in the form of Types. These will act as constraints on the implementation space, bringing with them reusable and tested code.`